alibuild_helpers/build.py

```diff
@@ -344,33 +344,30 @@ def doBuild(args, parser):
                   "  git pull --rebase\n",
                   pwd=os.getcwd(), star=star()))
 
-  # Prefetch git heads in parallel
-  debug("Prefetching git heads in parallel:")
-
-  fetchers = list(filter(lambda p: "source" in specs[p], buildOrder))
-  poolConfig = {"processes": min(max(len(fetchers), 1), 16)}
-  pool = ThreadPool(**poolConfig)
-  debug("Created thread pool with config: %s" % poolConfig)
-
-  git_commands = {} # store futures
-  for p in fetchers:
-    # Replace source with local one if we are in development mode.
+  # Clone/update repos
+  reposWithSource = list(filter(lambda p: "source" in specs[p], buildOrder))
+  if args.fetchRepos:
```

Comment:
This whole `if...else` can be simplified I believe (also, use `join` not `%s/%s`):

```python
for p in list(filter(lambda p: "source" in specs[p], buildOrder)):
  if not args.fetchRepos:
    specs[p]["reference"] = os.path.join(args.referenceSources, p.lower())
  if args.fetchRepos or not exists(spec[p]["reference"]):
    updateReferenceRepos(args.referenceSources, p, specs[p])
```

Reply:
List comprehensions for the win!
```python
for p in [p for p in buildOrder if "source" in specs[p]]:
```

Reply:
👍!